src/server/saved_objects/saved_objects_mixin.js

```diff
@@ -26,49 +26,17 @@ export function savedObjectsMixin(kbnServer, server) {
   server.route(createGetRoute(prereqs));
   server.route(createUpdateRoute(prereqs));
 
-  async function onBeforeWrite() {
-    const adminCluster = server.plugins.elasticsearch.getCluster('admin');
-
-    try {
-      const index = server.config().get('kibana.index');
-      await adminCluster.callWithInternalUser('indices.putTemplate', {
-        name: `kibana_index_template:${index}`,
-        body: {
-          template: index,
-          settings: {
-            number_of_shards: 1,
-            auto_expand_replicas: '0-1',
-          },
-          mappings: server.getKibanaIndexMappingsDsl(),
-        },
-      });
-    } catch (error) {
-      server.log(['debug', 'savedObjects'], {
-        tmpl: 'Attempt to write indexTemplate for SavedObjects index failed: <%= err.message %>',
-        es: {
-          resp: error.body,
-          status: error.status,
-        },
-        err: {
-          message: error.message,
-          stack: error.stack,
-        },
-      });
-
-      // We reject with `es.ServiceUnavailable` because writing an index
-      // template is a very simple operation so if we get an error here
-      // then something must be very broken
-      throw new adminCluster.errors.ServiceUnavailable();
+  let savedObjectsService;
+  server.decorate('server', 'savedObjects', () => {
```

Comment:
Mine elaborating on why this is a getter and not just a value? `server.decorate('server', 'savedObjects', createSavedObjectsService(server))` seems more obvious and doesn't require calling `server.savedObjects()` every time you need access to the service.

Reply:
I originally had it as you've described, and prefer it that way; however, the kbnTestServer used 
 in src/server/http/version_check.test.js and src/server/http/xsrf.test.js stopped working because `createSavedObjectsService` was calling `config.get('kibana.index')` when this didn't exist and causing issues. I assumed this was why `server.config()` was done this way, and tried to follow suit. If that wasn't a conscious design decision, I can look into fixing the tests to work with this pattern.